Content.Server/Power/EntitySystems/BatterySystem.cs

```diff
@@ -110,6 +121,13 @@ public float UseCharge(EntityUid uid, float value, BatteryComponent? battery = n
             var newValue = Math.Clamp(0, battery.CurrentCharge - value, battery.MaxCharge);
             var delta = newValue - battery.CurrentCharge;
             battery.CurrentCharge = newValue;
+
+            if (TryComp<BatterySelfRechargerComponent>(uid, out var batterySelfRechargerComponent)){
+                if (batterySelfRechargerComponent.AutoRechargePause && batterySelfRechargerComponent.AutoRechargePauseTime > 0 && value > 0) {
+                    SetChargeCooldown(uid, Math.Clamp(batterySelfRechargerComponent.AutoRechargeCooldown, batterySelfRechargerComponent.AutoRechargePauseTime, 9999), batterySelfRechargerComponent);
+                }
+            }
+
```

Comment:
This formatting's really not great you're also double-clamping the value, once here and once in setchargecooldown. Value is also already checked 4 lines above so is pointless here.

Also please document it.

Reply:
Fixed,